charts/kubeclarity/templates/clusterrole.yaml

```diff
@@ -9,9 +9,11 @@ rules:
   - apiGroups: [""]
     resources: ["configmaps"]
     verbs: ["get"]
+  {{- if index .Values "clusterRole" "readClusterSecrets" }}
```

Comment:
Did you run KubeClarity after removing it? Didn't you got some failures when trying to read secrets? I think that there is a code change needed, no?

Reply:
@FrimIdan we don't need that in our cluster since we give the ECR Credentials via the EKS NodeRole.
Giving Cluster-Wide Secrets Read Privileges is problematic
BTW, this was my 1st motivation...and I'm sure other organizations would not go with that!

Reply:
@yossicohn I understand it, I just think that the current logic in the code assume that there kubeclariy have permissions to `get` secrets, so once you remove it from here while running kubeclarity you will get some failures when trying to get secrets.
For example the code in https://github.com/openclarity/kubeclarity/blob/main/runtime_scan/pkg/scanner/creds/helpers.go#L27-L39 is used during the creation of the scanner job in https://github.com/openclarity/kubeclarity/blob/main/runtime_scan/pkg/scanner/job_managment.go#L299-L304

Can you get the logs from kubeclarity during a scan when you are running it w/o the get secrets permissions?

Reply:
@FrimIdan I do see these error logs...though as explained since we have the NodesRole on our EKS Nodes we have credentials to get to the ECR Images.
So there are error logs but no problems
**Logs from KubeClarity Pod during scan**
`
2023/01/08 11:45:34 /build/backend/pkg/database/scheduler.go:58 record not found
[0.803ms] [rows:0] SELECT * FROM "scheduler" ORDER BY "scheduler"."id" LIMIT 1
2023/01/08 11:45:34 Serving kube clarity a p is at http://[::]:8080
2023/01/08 11:45:34 Serving kube clarity runtime scan a p is at http://[::]:8888
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:44Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:45Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:45Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
I0108 13:14:45.533535       1 request.go:682] Waited for 1.188134405s due to client-side throttling, not priority and fairness, request: GET:https://172.20.0.1:443/api/v1/namespaces/kubeclarity/secrets/ecr-sa
time="2023-01-08T13:14:45Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:45Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:45Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:46Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:46Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:46Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:46Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:46Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:47Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:47Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-08T13:14:51Z" level=warning msg="Vulnerabilities scan of imageID \"749425658711.dkr.ecr.us-east-1.amazonaws.com/hs-account-settings@sha256:622a4dc610611545ed348909aedc3121605bf43d4f33ea6d698d18a906ef145d\" has failed: &{failed to analyze image: failed to run job manager: failed to run job: failed to create source analyzer=syft: could not fetch image \"749425658711.dkr.ecr.us-east-1.amazonaws.com/hs-account-settings@sha256:622a4dc610611545ed348909aedc3121605bf43d4f33ea6d698d18a906ef145d\": unable to use OciRegistry source: unable to parse registry reference=\"registry:749425658711.dkr.ecr.us-east-1.amazonaws.com/hs-account-settings@sha256:622a4dc610611545ed348909aedc3121605bf43d4f33ea6d698d18a906ef145d\": could not parse reference: registry:749425658711.dkr.ecr.us-east-1.amazonaws.com/hs-account-settings@sha256:622a4dc610611545ed348909aedc3121605bf43d4f33ea6d698d18a906ef145d TBD}" func="github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner.(*Scanner).HandleScanResults" file="/build/runtime_scan/pkg/scanner/scanner.go:400" scanner id=79b4a5c0-5808-4f28-b710-9986af4fa4fd

2023/01/08 13:16:49 /build/backend/pkg/database/application.go:236 record not found
[1.144ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = '74c7aa9f-f859-5b69-99f0-9384f016b742' ORDER BY "applications"."id" LIMIT 1

2023/01/08 13:16:52 /build/backend/pkg/database/application.go:236 record not found
[1.354ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = 'b8e6730f-4e19-584a-8d2f-762c11b28284' ORDER BY "applications"."id" LIMIT 1

2023/01/08 13:16:55 /build/backend/pkg/database/application.go:236 record not found
[0.289ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = '93678100-43d2-59d9-8053-5d81cc68ceb2' ORDER BY "applications"."id" LIMIT 1
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=basic-regcred: secrets \"basic-regcred\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:56Z" level=error msg="Failed to get secret. secret=ecr-sa: secrets \"ecr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:57Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:57Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
I0110 10:28:57.408201       1 request.go:682] Waited for 1.187427968s due to client-side throttling, not priority and fairness, request: GET:https://172.20.0.1:443/api/v1/namespaces/kubeclarity/secrets/gcr-sa
time="2023-01-10T10:28:57Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:57Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:57Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"
time="2023-01-10T10:28:58Z" level=error msg="Failed to get secret. secret=gcr-sa: secrets \"gcr-sa\" is forbidden: User \"system:serviceaccount:kubeclarity:hs-kubeclarity\" cannot get resource \"secrets\" in API group \"\" in the namespace \"kubeclarity\"" func=github.com/openclarity/kubeclarity/runtime_scan/pkg/scanner/creds.isSecretExists file="/build/runtime_scan/pkg/scanner/creds/helpers.go:34"

2023/01/10 10:29:07 /build/backend/pkg/database/refresh_materialized_views.go:155 SLOW SQL >= 200ms
[227.801ms] [rows:0] REFRESH MATERIALIZED VIEW CONCURRENTLY packages_view;

2023/01/10 10:29:07 /build/backend/pkg/database/refresh_materialized_views.go:155 SLOW SQL >= 200ms
[317.380ms] [rows:0] REFRESH MATERIALIZED VIEW CONCURRENTLY vulnerabilities_view;

2023/01/10 10:29:12 /build/backend/pkg/database/refresh_materialized_views.go:155 SLOW SQL >= 200ms
[206.474ms] [rows:0] REFRESH MATERIALIZED VIEW CONCURRENTLY vulnerabilities_view;

2023/01/10 10:29:22 /build/backend/pkg/database/refresh_materialized_views.go:155 SLOW SQL >= 200ms
[229.563ms] [rows:0] REFRESH MATERIALIZED VIEW CONCURRENTLY vulnerabilities_view;

2023/01/10 10:29:23 /build/backend/pkg/database/application.go:236 record not found
[3.262ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = '2ecf5961-655a-5919-a328-319bd9ceb7f1' ORDER BY "applications"."id" LIMIT 1

2023/01/10 10:29:23 /build/backend/pkg/database/application.go:236 record not found
[0.400ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = 'b1ef7cc0-e24b-5bc4-8f13-71ae3eb793fd' ORDER BY "applications"."id" LIMIT 1

2023/01/10 10:29:23 /build/backend/pkg/database/application.go:236 record not found
[0.344ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = '6d5b618b-980f-5613-acf6-c01574462f65' ORDER BY "applications"."id" LIMIT 1

2023/01/10 10:29:24 /build/backend/pkg/database/application.go:236 record not found
[0.382ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = '3b6c0a61-d379-5830-a775-d2b4c86484c8' ORDER BY "applications"."id" LIMIT 1

2023/01/10 10:29:24 /build/backend/pkg/database/application.go:236 record not found
[0.460ms] [rows:0] SELECT * FROM "applications" WHERE applications.id = '51525d98-0e51-5b3b-8dfa-29e33fe9111a' ORDER BY "applications"."id" LIMIT 1

2023/01/10 10:29:27 /build/backend/pkg/database/refresh_materialized_views.go:155 SLOW SQL >= 200ms
[202.376ms] [rows:0] REFRESH MATERIALIZED VIEW CONCURRENTLY vulnerabilities_view;
`

Reply:
@yossicohn I understand your case, I just think that we should avoid trying to look for secrets in case the permissions are removed (`readClusterSecrets == false`). 
Need to define an env var in kubeclarity that will be set when `readClusterSecrets` is `false` so that `CreateScanner` will not set `credentialAdders`. WDYT?

Reply:
@FrimIdan I totally agree, this would not look healthy.
I can add the env var (`READ_CLUSTER_SECRETS`)to the `values.yaml` and `deployment.yaml`

Regarding the code AFAIU, correct me if I'm wrong in the scanner
[scanner.go](https://github.com/openclarity/kubeclarity/blob/8fb35ff0ddd586f592aac67433514d64508193f9/runtime_scan/pkg/scanner/scanner.go#LL62C21-L62C46)
In the case of no secrets privileges, we need to have `credentialAdders` set to empty `[]_creds.CredentialAdder` array?

`
credentialAdders: []_creds.CredentialAdder{}
`

Reply:
@yossicohn that is correct.